test/lib/jdk/test/lib/process/OutputAnalyzer.java

```diff
@@ -176,13 +178,15 @@ public OutputAnalyzer stderrShouldBeEmptyIgnoreWarnings() {
 
     /**
      * Verify that the stderr contents of output buffer is empty,
-     * after filtering out the Hotspot deprecation warning messages
+     * after filtering out the Hotspot deprecation warning messages,
+     * and warning messages about removal.
      *
      * @throws RuntimeException
      *             If stderr was not empty
      */
     public OutputAnalyzer stderrShouldBeEmptyIgnoreDeprecatedWarnings() {
```

Comment:
Maybe this should be renamed to reflect that it is about VM option warnings.

Reply:
Possibly. It means updating 15 tests. Also need to come up with a new name. Any suggestions?

Reply:
Hello Chris, given these similary named methods in this class, perhaps we should instead just have one single `stderrShouldBeEmptyIgnoring(...)` method which takes the messages that should be ignored. Something like:

```java
public static final String VM_WARNING_MSG = ".* VM warning:.*";

public static final String VM_WARNING_DEPRECATED_MSG = ".* VM warning:.* deprecated.*";
...

public OutputAnalyzer stderrShouldBeEmptyIgnoring(String ignoreMsg, String... additionalIgnoreMsgs) {
    String stdErrContent = getStderr().replaceAll(ignoreMsg + "\\R", "");
    if (additionalIgnoreMsgs != null) {
    	for (String additionalIgnore : additionalIgnoreMsgs) {
    		stdErrContent = getStderr().replaceAll(additionalIgnore + "\\R", "");
    	}
    }
    if (!stdErrContent.isEmpty()) {
    	reportDiagnosticSummary();
    	throw new RuntimeException("stderr was not empty");
    }
    return this;
}

```
We make those private fields in OutputAnalyzer public and have the caller pass them:

```
oa.stderrShouldBeEmptyIgnoring(OutputAnalyzer.VM_WARNING_DEPRECATED_MSG)
```

Reply:
That seems like a lot of busy work and over abstraction for trying to solve a rather trivial issue that already has a very simple solution, but maybe could use a better API name.

Reply:
In that case, would `stderrShouldBeEmptyIgnoreVMOptionDeprecations` be OK?